modules/core/src/main/java/org/apache/synapse/mediators/MediatorLog.java

```diff
@@ -114,8 +121,10 @@ public void auditLog(Object msg) {
      * Log a message at level DEBUG to all available/enabled logs.
      */
     public void auditDebug(Object msg) {
-        defaultLog.debug(msg);
-        if (synCtx.getServiceLog() != null) {
+    	if (defaultLog.isDebugEnabled()) { 
```

Comment:
Yes, you need this defaultLog.debug(msg) part. In standard practice, isDebugEnabled is used to avoid unnecessary string processing before calling actual debug() method. If there are no such string processing, having an if() clause is a waste.
Anyway, this is a just a minor concern :) I'll merge this PR!